Docs: moduleResolution change to "bundler" is a breaking change not clearly documented in upgrade guide

### Describe the problem that you experienced

Upgrading to Angular 21 introduced unexpected TypeScript and build issues due to an automatic migration of moduleResolution from "node" to "bundler" (via https://github.com/angular/angular/pull/64125
).

The issue is that this behavior is not clearly documented as a breaking change in the Angular upgrade guide, making it difficult to diagnose the root cause.

### Enter the URL of the topic with the problem

https://angular.dev/update-guide?v=20.0-21.0&l=3

### Describe what you were looking for in the documentation

I expected to find a clear mention in the Angular 21 upgrade guide that:

* moduleResolution is changed to "bundler"
* This change may introduce breaking behavior
* Guidance on how to handle issues caused by this change

### Describe the actions that led you to experience the problem

_No response_

### Describe what you want to experience that would fix the problem

_No response_

### Add a screenshot if that helps illustrate the problem

_No response_

### If this problem caused an exception or error, please paste it here

```true

```

### If the problem is browser-specific, please specify the device, OS, browser, and version

```true

```

### Provide any additional information here in as much as detail as you can

```true

```

Hi Alan :)

So we have a very large nx monorepo, with multiple angular apps, libs, etc..
our base tsconfig is defined to `"moduleResolution": "node"`
(we are working on a migration to `bundler` these days, regardless of the angular update)

when I tried to update angular, after installing v21, things that have secondary entry-points were failed to work

examples:
```
import {takeUntilDestroyed} from '@angular/core/rxjs-interop';   

error TS2307: Cannot find module '@angular/core/rxjs-interop' or its corresponding type declarations.                                                        
```




angular v20 had a backwards compability, meaning it had a folder named `rxjs-interop` inside the `node_modules/@angular/core/rxjs-interop` path
angular v21 no longer have this folder, and it must be used in a `bundler` configuration according to my understanding.

now regarding the breaking changes -
if I switch to `bundler` our configuration, it fixes the above error, however many other things in our monorepo that relayed on `node` resolution is now throw errors and we need to address them.

as I said, its something we planned to do anyway, but now its more urgent since its prerequisites to angular v21

All I wanted is to make this change more visible to others by mentioning it in the update guide, since v20 had backwards compatibility and v21 doesn't

let me know what do you think.

--- a/adev/src/app/features/update/recommendations.ts
+++ b/adev/src/app/features/update/recommendations.ts
@@ -2706,6 +2706,14 @@ export const RECOMMENDATIONS: Step[] = [
   {
     possibleIn: 2000,
     necessaryAsOf: 2000,
+    level: ApplicationComplexity.Medium,
+    step: '20.0.0_set_moduleResolution_to_bundler',
+    action:
+      "Set `moduleResolution` to `'bundler'` in your `tsconfig.json`. Angular CLI's `ng update` migration applies this change automatically; if you upgrade manually or override the option in a base tsconfig, set it explicitly so imports of secondary entry-points such as `@angular/core/rxjs-interop` continue to resolve correctly.",
+  },
+  {
+    possibleIn: 2000,
+    necessaryAsOf: 2000,
     level: ApplicationComplexity.Advanced,
     step: '20.0.0_review_AsyncPipe_error_handling_in_tests',
     action:

